src/jaxsim/mujoco/utils.py

```diff
@@ -0,0 +1,105 @@
+            model reduction process will be set to their initial values.
+
+    Returns:
+        The Mujoco data object containing the state of the JaxSim model.
+
+    Note:
+        This method is useful to initialize a Mujoco data object used for visualization
+        with the state of a JaxSim model. In particular, this function takes care of
+        initializing the positions of the joints that have been removed during the
+        model reduction process. After the initial creation of the Mujoco data object,
+        it's faster to update the state using an external MujocoModelHelper object.
+    """
+
+    # The package `jaxsim.mujoco` is supposed to be jax-independent.
+    # We import all the JaxSim resources privately.
+    import jaxsim.api as js
+
+    if not isinstance(jaxsim_model, js.model.JaxSimModel):
+        raise ValueError("The `jaxsim_model` argument must be a JaxSimModel object.")
+
+    if not isinstance(jaxsim_data, js.data.JaxSimModelData):
+        raise ValueError("The `jaxsim_data` argument must be a JaxSimModelData object.")
+
+    # Create the helper to operate on the Mujoco model and data.
+    model_helper = MujocoModelHelper(model=mujoco_model, data=mujoco_data)
+
+    # Set the model position.
+    if jaxsim_model.floating_base():
+
+        model_helper.set_base_position(position=np.array(jaxsim_data.base_position()))
+
+    # Set the model orientation.
+    if jaxsim_model.floating_base():
+
+        model_helper.set_base_orientation(
+            orientation=np.array(jaxsim_data.base_orientation())
+        )
+
+    # Set the joint positions.
+    if jaxsim_model.dofs() > 0:
+
+        model_helper.set_joint_positions(
+            joint_names=list(jaxsim_model.joint_names()),
+            positions=np.array(
+                jaxsim_data.joint_positions(
+                    model=jaxsim_model, joint_names=jaxsim_model.joint_names()
+                )
+            ),
+        )
+
+    # Updating these joints is not necessary after the first time.
+    # Users can disable this update after initialization.
+    if update_removed_joints:
+
+        # Create a dictionary with the joints that have been removed for various reasons
+        # (like link lumping due to model reduction).
+        joints_removed_dict = {
+            j.name: j
+            for j in jaxsim_model.description._joints_removed
+            if j.name not in set(jaxsim_model.joint_names())
+        }
+
+        # Get all the joints in the Mujoco and JaxSim models.
+        joints_mujoco = set(model_helper.joint_names())
+
+        # Set the positions of the removed joints.
+        for joint_name in set(joints_removed_dict.keys()):
+
+            if joint_name not in joints_mujoco:
+                continue
+
+            _ = [
+                model_helper.set_joint_position(
+                    position=joints_removed_dict[joint_name].initial_position,
+                    joint_name=joint_name,
+                )
+            ]
```

Comment:
`set_joint_position` returns `None`
```suggestion
            model_helper.set_joint_position(
                position=joints_removed_dict[joint_name].initial_position,
                joint_name=joint_name,
            )
```

Reply:
This is likely a leftover of a missing refactoring to remove the for loop. Done in https://github.com/ami-iit/jaxsim/pull/264/commits/573b35a3bca6ea17c18d301eb5305ab17e2265ce.